pkgs/data/themes/catppuccin-gtk/default.nix

```diff
@@ -4,64 +4,81 @@
 , gtk3
 , gnome-themes-extra
 , gtk-engine-murrine
+, python3
 , sassc
-, tweaks ? [ ]
+, accent ? "blue"
 , size ? "standard"
+, tweaks ? [ ]
+, variant ? "frappe"
 }:
 let
+  validAccents = [ "blue" "flamingo" "green" "lavender" "maroon" "mauve" "peach" "pink" "red" "rosewater" "sapphire" "sky" "teal" "yellow" ];
   validSizes = [ "standard" "compact" ];
-  validTweaks = [ "nord" "dracula" "black" "rimless" "normal" ];
-
-  unknownTweaks = lib.subtractLists validTweaks tweaks;
-  illegalMix = (lib.elem "nord" tweaks) && (lib.elem "dracula" tweaks);
+  validTweaks = [ "black" "rimless" "normal" ];
+  validVariants = [ "latte" "frappe" "macchiato" "mocha" ];
 
-  assertIllegal = lib.assertMsg (!illegalMix) ''
-    Tweaks "nord" and "dracula" cannot be mixed. Tweaks: ${toString tweaks}
+  assertAccent = lib.assertMsg (lib.elem accent validAccents) ''
+    You entered an invalid accent: ${accent}
+    Valid accents are: ${toString validAccents}
   '';
 
   assertSize = lib.assertMsg (lib.elem size validSizes) ''
-    You entered a wrong size: ${size}
+    You entered an invalid size: ${size}
     Valid sizes are: ${toString validSizes}
   '';
 
-  assertUnknown = lib.assertMsg (unknownTweaks == [ ]) ''
-    You entered wrong tweaks: ${toString unknownTweaks}
+  unknownTweaks = lib.subtractLists validTweaks tweaks;
+  assertTweaks = lib.assertMsg (unknownTweaks == [ ]) ''
+    You entered invalid tweaks: ${toString unknownTweaks}
     Valid tweaks are: ${toString validTweaks}
   '';
+
+  assertVariants = lib.assertMsg (lib.elem variant validVariants) ''
+    You entered an invalid variant: ${variant}
+    Valid variants are: ${toString validVariants}
+  '';
 in
 
-assert assertIllegal;
+assert assertAccent;
 assert assertSize;
-assert assertUnknown;
+assert assertTweaks;
+assert assertVariants;
```

Comment:
Why do we set the variable in the let in and assert here?

Reply:
That's how it's been done since the existence of asserts in this package. If there's a better way, can you point me to it so I can update this?